.github/workflows/update_collector_periodic.yaml

```diff
@@ -4,21 +4,46 @@ on:
   schedule:
     - cron: 0 5 * * 1-5
 jobs:
-  update-collector:
+  determine-new-collector-version:
     runs-on: ubuntu-latest
+    outputs:
+      new-collector-version: ${{ steps.update-version.outputs.new-collector-version }}
     steps:
-    - uses: actions/checkout@v4
-      with:
-        ref: master # this is our target branch
-        fetch-depth: 0 # we need to fetch all branches
     - name: Checkout collector repo
       uses: actions/checkout@v4
       with:
         repository: stackrox/collector
-        path: deps/collector
         fetch-depth: 0 # we need to fetch tags
     - name: Update version
-      run: make -sC deps/collector tag | tee COLLECTOR_VERSION
+      id: update-version
+      run: |
+        echo "new-collector-version=$(make -s tag)" >> "${GITHUB_OUTPUT}"
+
+  ensure-images-exist:
+    runs-on: ubuntu-latest
+    needs: determine-new-collector-version
+    strategy:
+      matrix:
+        tag_suffix: ["", "-fast"]
+    steps:
+    - name: Ensure image exists
+      uses: stackrox/actions/release/wait-for-image@v1
+      with:
+        token: ${{ secrets.QUAY_RHACS_ENG_BEARER_TOKEN }}
+        image: rhacs-eng/collector:${{ needs.determine-new-collector-version.outputs.new-collector-version }}${{ matrix.tag_suffix }}
+        limit: 1800
```

Comment:
What happens when this fails? Does the scheduled workflow get quietly failing without notifying anyone via Slack and without updating the existing PRs or opening the new ones?

If yes, it would be hard to notice when PRs stop updating in the volume of all other auto-generated PRs.

Also, this check here doesn't catch when someone manually updates SCANNER_VERSION/COLLECTOR_VERSION in their PR.

I'd suggest to introduce the check for images existence in one of the PR/test workflows and have it conditionally enabled only when SCANNER_VERSION or COLLECTOR_VERSION get changed.

This way we both catch the case when PRs opened by this automation try bump pointers to commits for which Konflux builds failed and also prevent manual updates to broken builds, and the check status will be visible to engineers.

WDYT?

Reply:
Moved the check to `build.yaml` as `check-collector-images-exist ` and `check-scanner-images-exist` - as a prerequisite for the image build. 
This will be reported as part of the build pipeline (which is a required check) either as a failed check on a PR or as a Slack message on pushes with the existing mechanisms.